src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1433,34 +1434,13 @@ private static string GetInputTypeString(InputType inputType)
             return selectList;
         }
 
-        private static IEnumerable<SelectListItem> UpdateSelectListItemsWithDefaultValue(
-            ModelExplorer modelExplorer,
-            IEnumerable<SelectListItem> selectList,
-            ICollection<string> currentValues)
+        /// <inheritdoc />
+        public IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<SelectListItem> selectList)
         {
-            // Perform deep copy of selectList to avoid changing user's Selected property values.
-            var newSelectList = new List<SelectListItem>();
-            foreach (SelectListItem item in selectList)
-            {
-                var value = item.Value ?? item.Text;
-                var selected = currentValues.Contains(value);
-                var copy = new SelectListItem
-                {
-                    Disabled = item.Disabled,
-                    Group = item.Group,
-                    Selected = selected,
-                    Text = item.Text,
-                    Value = item.Value,
-                };
-
-                newSelectList.Add(copy);
-            }
-
-            return newSelectList;
+            return GenerateGroupsAndOptions(optionLabel: optionLabel, selectList: selectList, currentValues: null);
         }
 
-        /// <inheritdoc />
-        public IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<SelectListItem> selectList)
+        private IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<SelectListItem> selectList, ICollection<string> currentValues)
```

Comment:
- Wrap this line w/ each parameter on its own line.
- `currentValues` should be either `HashSet<string>` or `ISet<string>` to avoid Linq use.